libs/client/src/types.ts

```diff
@@ -33,3 +33,9 @@ export type QueueStatus =
 export function isQueueStatus(obj: any): obj is QueueStatus {
```

Comment:
Does this have to be ambiguous?

Reply:
This is a type-predicate (formerly known as type-guard). So the function returns true/false during the runtime, but in build-time it gives the compiler information about the type, so:

```
const someUntypedJsonComingFromARequest = /* ... */
if (isQueueStatus(someUntypedJsonComingFromARequest)) {
  // now the compiler knows that someUntypedJsonComingFromARequest  is actually QueueStatus
}
```

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

---

Does that clarify it?